app/src/main/java/com/github/sdp_begreen/begreen/activities/SignInActivity.kt

```diff
@@ -93,13 +99,43 @@ class SignInActivity : AppCompatActivity() {
                     if (it.isNotBlank()) {
                         checkUserExistence()
                         startActivity(Intent(this, MainActivity::class.java))
-                        // Here I will hide the progress bar when implemented
+                        hideProgressDialog()
                         finish()
                     }
                 }
             }
     }
 
+    // Variable to hold the progress dialog
+    private var dialog: AlertDialog? = null
+
+    /**
+     * Displays a progress dialog with a loading circle while a process is executing.
+     *
+     * @param context The context in which the progress dialog will be displayed.
+     */
+    private fun showProgressDialog(context: Context) {
+        if (dialog?.isShowing == true) return // Dialog is already displayed, no need to create a new one.
+
+        dialog = AlertDialog.Builder(context)
+            .setView(R.layout.loading_circle)
+            .setCancelable(false)
+            .create().apply {
+                window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+                window?.clearFlags(WindowManager.LayoutParams.FLAG_DIM_BEHIND)
+                show()
+            }
+    }
+
+
+    /**
+     * Hides the progress dialog.
+     */
+    private fun hideProgressDialog() {
```

Comment:
You use two different approaches for the show and hide progress dialog. For better clarity, change this so its coherent with the _showProgressDialog_ :

```
if (dialog?.isShowing == false) {
            return
}


dialog!!.dismiss()

dialog = null```

Reply:
Thanks for the proposition. However, I am not certain that this line below represents best Android practice : 

dialog!!.dismiss()

Those two methods are in fact quite different therefore I don't think that is very useful to increase the number of line codes of hideProgressDialog() for the sole purpose to be more coherent with ShowProgressDialog. 

If my explanation is ok for you, I'd prefer to leave it as it is. What do you think ?